rust/src/format/manifest.rs

```diff
@@ -44,16 +46,29 @@ pub struct Manifest {
 
     /// The file position of the index metadata.
     pub index_section: Option<usize>,
+
+    /// The creation timestamp with nanosecond resolution as 128-bit integer
+    pub timestamp_nanos: u128,
+
+    /// An optional string tag for this version
+    pub tag: Option<String>,
 }
 
 impl Manifest {
     pub fn new(schema: &Schema, fragments: Arc<Vec<Fragment>>) -> Self {
+        let duration_since_epoch = SystemTime::now()
```

Comment:
This timestamp should be generated right before the commit (where the manifest is written AFTER all data is written).